packages/amplify-category-storage/src/provider-utils/awscloudformation/service-walkthroughs/s3-user-input-state.ts

```diff
@@ -0,0 +1,162 @@
+    schemaValidator.validateInput(JSON.stringify(this._inputPayload!));
+  }
+
+  public static getPermissionTypeFromCfnType(s3pPrmissionCfnType: S3CFNPermissionType): S3PermissionType {
+    switch (s3pPrmissionCfnType) {
+      case S3CFNPermissionType.PUT_OBJECT:
+        return S3PermissionType.PUT_OBJECT;
+      case S3CFNPermissionType.GET_OBJECT:
+        return S3PermissionType.GET_OBJECT;
+      case S3CFNPermissionType.DELETE_OBJECT:
+        return S3PermissionType.DELETE_OBJECT;
+      case S3CFNPermissionType.LIST_BUCKET:
+        return S3PermissionType.LIST_BUCKET;
+      default:
+        throw new Error(`Unknown CFN Type: ${s3pPrmissionCfnType}`);
+    }
+  }
+
+  public static getInputPermissionsFromCfnPermissions(selectedGuestPermissions: S3CFNPermissionType[] | undefined) {
+    if (selectedGuestPermissions) {
+      return selectedGuestPermissions.map(S3InputState.getPermissionTypeFromCfnType);
+    } else {
+      return [];
+    }
+  }
+
+  public static cliWalkThroughToCliInputParams(cliInputsFilePath: string, options: S3CLIWalkthroughParams) {
+    const inputProps: S3InputStateOptions = {
+      userInputDataFilename: cliInputsFilePath,
+      resourceName: options.resourceName,
+      inputPayload: {
+        resourceName: options.resourceName,
+        bucketName: options.bucketName,
+        storageAccess: options.storageAccess,
+        selectedGuestPermissions: S3InputState.getInputPermissionsFromCfnPermissions(options.selectedGuestPermissions),
+        selectedAuthenticatedPermissions: S3InputState.getInputPermissionsFromCfnPermissions(options.selectedAuthenticatedPermissions),
+        isTriggerEnabled: options.triggerFunction !== 'NONE' ? true : false, //enable if trigger
+        triggerFunctionName: options.triggerFunction !== 'NONE' ? options.triggerFunction : undefined,
+      },
+    };
+    return inputProps;
+  }
+
+  updateInputPayload(props: S3InputStateOptions) {
+    // Overwrite
+    this._inputPayload = props.inputPayload;
+
+    // validate cli-inputs.json
+    const schemaValidator = new CLIInputSchemaValidator(this._service, this._category, 'S3UserInputs');
+    schemaValidator.validateInput(JSON.stringify(this._inputPayload!));
+  }
+
+  public static getInstance(props: S3InputStateOptions): S3InputState {
+    const projectPath = pathManager.findProjectRoot();
+    if (!S3InputState.s3InputState) {
+      S3InputState.s3InputState = new S3InputState(props);
+    }
+    S3InputState.s3InputState.updateInputPayload(props);
+    return S3InputState.s3InputState;
+  }
+
+  public getCliInputPayload(): S3UserInputs {
+    if (this._inputPayload) {
+      return this._inputPayload;
+    } else {
+      throw new Error('cli-inputs not present. Either add category or migrate project to support extensibility');
+    }
+  }
+
+  public saveCliInputPayload(): void {
+    const backend = pathManager.getBackendDirPath();
```

Comment:
nit: use pathManager.getResourceDirectoryPath()

Reply:
@sachscode ^^ just for reference for your PR.